mypy/types_utils.py

```diff
@@ -0,0 +1,160 @@
+"""
+This module is for (more basic) type operations that should not depend on is_subtype(),
+meet_types(), join_types() etc. We don't want to keep them in mypy/types.py for two reasons:
+* Reduce the size of that module.
+* Reduce use of get_proper_type() in types.py to avoid cyclic imports
+  expand_type <-> types, if we move get_proper_type() to the former.
+"""
+
+from __future__ import annotations
+
+from typing import Callable, Iterable, cast
+
+from mypy.nodes import ARG_STAR, ARG_STAR2, FuncItem, TypeAlias
+from mypy.types import (
+    AnyType,
+    CallableType,
+    Instance,
+    NoneType,
+    Overloaded,
+    ParamSpecType,
+    ProperType,
+    TupleType,
+    Type,
+    TypeAliasType,
+    TypeType,
+    TypeVarType,
+    UnionType,
+    UnpackType,
+    flatten_nested_unions,
+    get_proper_type,
+    get_proper_types,
+)
+
+
+def flattened(types: Iterable[Type]) -> Iterable[Type]:
```

Comment:
If we're moving this anyway, `flatten_types` might be a better, more descriptive name here